site-src/geps/gep-1709.md

```diff
@@ -37,8 +45,295 @@ which implementations can prove they satisfy and be recognized for.
 
 ## API
 
-TODO
+The API for conformance profiles will be a pipeline that implementations can
+opt into. The workflow is effectively:
+
+1. select a [profile](#profiles)
+2. [integrate](#integration) tests in the downstream project
+3. [report results and get certified](#certification)
+
+The goal is to make selecting a conformance profile as simple of a process as
+feasible and support both the existing command line integration approach as well
+as a [Golang][go] approach using the conformance suite as a library.
+
+[go]:https://go.dev
+
+### Profiles
+
+Initially there will be three _categories_ of profiles which can be coupled with
+two _support levels_. These will be identified by short [CamelCase][camelcase]
+names. The initial categories we'll support will be:
+
+- `Layer4`
+- `Layer7`
+- `Mesh`
+
+> **NOTE**: these are simply the initial categories we're going to start with,
+> it's entirely reasonable for there to be more in the future.
+
+Support levels will be limited to `Core` and `Extended`, since
+`ImplementationSpecific` isn't meaningful in conformance at the time of
```

Comment:
There are some impls that support experimental features, like istio and envoy-gw. 
Whats the additional cost of providing the profile for them to be able to include these tests?

Reply:
In my first pass thinking through this I honestly didn't really think experimental features needed any special recognition, as they have a special nature that makes them off the normal course of the project's feature scope. That said, if there's some strong feelings we need to represent _experimental feature support_ visibly at a high level with conformance profiles I am open to that, I'm just not quite convinced yet :thinking:

Reply:
After talking with @LiorLieberman in a video call, the entire concept of how you refer to profiles has been reworked to incorporate providing a positive (or negative) list of additional features so this should be set now.